app/vendor/views.py

```diff
@@ -1,11 +1,116 @@
-from flask import render_template
-from . import vendor
 from ..decorators import vendor_required
-from flask.ext.login import login_required
+
+from flask import render_template, abort, redirect, flash, url_for
+from flask.ext.login import login_required, current_user
+
+from forms import (
+	ChangeListingInformation,
+	NewItemForm   
+)  
+from . import vendor
+from ..models import Listing, Category
+from .. import db
 
 
 @vendor.route('/')
 @login_required
 @vendor_required
 def index():
-    return render_template('main/index.html')
+    return render_template('vendor/index.html')
+
+@vendor.route('/new-item', methods=['GET', 'POST'])
+@login_required
+@vendor_required
+def new_listing():
+    """Create a new item."""
+    form = NewItemForm()
+    if form.validate_on_submit():
+        category_id = form.categoryId.data.id
+        name = form.listingName.data
+        if current_user.listings.filter_by(name=name).first():
+            flash('Item {} already exists'.format(name), 'form-error')
+            return render_template('vendor/new_listing.html', form=form)
+        listing = Listing(
+                name=name,
+                description=form.listingDescription.data,
+                available=True,
+                price=form.listingPrice.data,
+                category_id=category_id,
+                vendor_id=current_user.id
+        )
+        db.session.add(listing)
+        db.session.commit()
+        flash('Item {} successfully created'.format(listing.name),
+              'form-success')
+    return render_template('vendor/new_listing.html', form=form)
+
+@vendor.route('/items')
+@login_required
+@vendor_required
+def current_listings():
+    """View all current listings."""
+    listings = Listing.query.all()
+    categories = Category.query.all()
+    return render_template('vendor/current_listings.html',
+                           listings=listings,
+                           categories=categories)
+
+
+@vendor.route('/items/<int:listing_id>')
+@vendor.route('/items/<int:listing_id>/info')
+@login_required
+@vendor_required
+def listing_info(listing_id):
+    """View a listing's info."""
+    listing = Listing.query.filter_by(id=listing_id).first()
+    if listing is None:
+        abort(404)
+    return render_template('vendor/manage_listing.html', listing=listing)
+
+
+@vendor.route('/items/<int:listing_id>/edit-item', methods=['GET', 'POST'])
+@login_required
+@vendor_required
+def change_listing_info(listing_id):
```

Comment:
Same as `delete_listing`. Plz check `listing.vendor_id == current_user.id` (`current_user is listing.vendor` might also work and be more idiomatic, I'm not sure).